TTS/tts/models/xtts.py

```diff
@@ -442,32 +442,57 @@ def get_conditioning_latents(
         librosa_trim_db=None,
         sound_norm_refs=False,
     ):
-        speaker_embedding = None
-        diffusion_cond_latents = None
-
-        audio, sr = torchaudio.load(audio_path)
-        audio = audio[:, : sr * max_ref_length].to(self.device)
-        if audio.shape[0] > 1:
-            audio = audio.mean(0, keepdim=True)
-        if sound_norm_refs:
-            audio = (audio / torch.abs(audio).max()) * 0.75
-        if librosa_trim_db is not None:
-            audio = librosa.effects.trim(audio, top_db=librosa_trim_db)[0]
-
-        if self.args.use_hifigan or self.args.use_ne_hifigan:
-            speaker_embedding = self.get_speaker_embedding(audio, sr)
+        # deal with multiples references
+        if not isinstance(audio_path, list):
+            audio_paths = list(audio_path)
         else:
-            diffusion_cond_latents = self.get_diffusion_cond_latents(audio, sr)
-        gpt_cond_latents = self.get_gpt_cond_latents(audio, sr, length=gpt_cond_len)  # [1, 1024, T]
-        return gpt_cond_latents, diffusion_cond_latents, speaker_embedding
+            audio_paths = audio_path
+
+        speaker_embeddings = []
+        diffusion_cond_latents = []
+        audios = []
+        speaker_embedding = None
+        diffusion_cond_latent = None
+        for file_path in audio_paths:
+            audio, sr = torchaudio.load(file_path)
```

Comment:
@erogol  Should we load the audio already in 24khz to prevent issues with references that have different sampling rate?

Reply:
yeah makes sense